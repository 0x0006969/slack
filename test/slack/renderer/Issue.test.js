const { Issue } = require('../../../lib/slack/renderer/issue');

const issuesOpened = require('../../fixtures/webhooks/issues.opened.json');
const issuesClosed = require('../../fixtures/webhooks/issues.closed.json');

describe('Issue rendering', () => {
  test('works for unfurls', async () => {
    const issueMessage = new Issue({
      issue: issuesOpened.issue,
      repository: issuesOpened.repository,
      unfurl: true,
    });
    const rendered = issueMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('works for notification messages', async () => {
    const issueMessage = new Issue({
      issue: issuesOpened.issue,
      repository: issuesOpened.repository,
      eventType: 'issues.opened',
    });
    const rendered = issueMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
  test('works for minor notifications', async () => {
    const issueMessage = new Issue({
      issue: issuesClosed.issue,
      repository: issuesClosed.repository,
      eventType: 'issues.closed',
    });
    const rendered = issueMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
});
