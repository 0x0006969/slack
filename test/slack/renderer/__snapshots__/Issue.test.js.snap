// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue rendering works for minor notifications 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#cb2431",
      "fallback": "[github-slack/public-test] Issue closed by wilhelmklopp",
      "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "pretext": "[github-slack/public-test] Issue closed by wilhelmklopp",
      "title": "#1 Needs content",
      "title_link": "https://github.com/github-slack/public-test/issues/1",
      "ts": 1508171038,
    },
  ],
}
`;

exports[`Issue rendering works for notification messages 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "[github-slack/public-test] Issue opened by wilhelmklopp",
      "fields": Array [
        Object {
          "short": true,
          "title": "Assignees",
          "value": "wilhelmklopp",
        },
        Object {
          "short": true,
          "title": "Labels",
          "value": "enhancement, question, wontfix",
        },
      ],
      "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "pretext": "[github-slack/public-test] Issue opened by wilhelmklopp",
      "text": "This repo needs some content, as currently, it does not have **any**.",
      "title": "#1 Needs content",
      "title_link": "https://github.com/github-slack/public-test/issues/1",
      "ts": 1508171038,
    },
  ],
}
`;

exports[`Issue rendering works for unfurls 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "#1 Needs content",
      "fields": Array [
        Object {
          "short": true,
          "title": "Assignees",
          "value": "wilhelmklopp",
        },
        Object {
          "short": true,
          "title": "Labels",
          "value": "enhancement, question, wontfix",
        },
      ],
      "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "This repo needs some content, as currently, it does not have **any**.",
      "title": "#1 Needs content",
      "title_link": "https://github.com/github-slack/public-test/issues/1",
      "ts": 1508171038,
    },
  ],
}
`;
